### DISCUSSION TITLE
How to enable the feature the "Street view"?

### DISCUSSION THREAD

**Discussion by kaiquefelipe** [visgl/react-google-maps#460] [Q&A]
How can I activate the “Street View” feature? to allow you to navigate the streets.
Can you help me?

Similar with the action the attach below.
![Captura de tela002800](https://github.com/user-attachments/assets/0cbbe8e1-880f-454c-90ac-44a99c253a34)

**Comment by usefulthink** [comment]
We haven't added any components or hooks for this yet. Can you describe your use-case (how will users interact with the StreetView feature) a bit more? Maybe we can provide some helpers for your use-case with the library.

Generally, all the features from the maps API are available to the react-library as well. So, you should be able to enable the PegMan (the yellow figure) button using the [`streetViewControl`](https://developers.google.com/maps/documentation/javascript/reference/map#MapOptions.streetViewControl) and [`streetViewControlOptions`](https://developers.google.com/maps/documentation/javascript/reference/map#MapOptions.streetViewControlOptions) props of the map component.

Programmatic access to StreetView will be possible on the map instance, using the `useMap`/`useEffect` hooks:

```tsx
const map = useMap();
useEffect(() => {
  if (!map) return;

  const sv = map.getStreetView();
  sv.setPosition(streetviewCoordinates);
  sv.setPov({ heading: 265, pitch: 0 });
  sv.setVisible(true);
}, [map]);
```
